### ISSUE TITLE
Duct Location Error

### ISSUE THREAD

**Issue by lukeduran** [RadiantLabs/modeling-api-docs#56]
We are getting several errors relating to Duct Location, but I can't seem to figure out what I can do to remedy these errors. The commonality between them all seems to be that we are passing a "Foundation" asset.

**Request**

```
{
            "addressFull": "218 S Ohio St, King City, MO 64463-9530",
            "addressComponents": {
                "addressLine1": "218 S Ohio St",
                "city": "King City",
                "stateAbbreviation": "MO",
                "zipCode": "64463-9530"
            },
            "baseBuilding": {
                "buildingSummary": {
                    "yearBuilt": 1929,
                    "bedroomsCount": 2,
                    "averageCeilingHeight": 8,
                    "conditionedFloorArea": 1077
                },
                "enclosure": {
                    "airInfiltration": {
                        "rate": 2313,
                        "rateUnit": "CFM"
                    },
                    "attics": [
                        {
                            "floorAssemblyEffectiveRValue": 38,
                            "id": "attic-insulation-1"
                        }
                    ],
                    "foundations": [
                        {
                            "type": "crawl unvented",
                            "id": "floor-and-foundation-insulation-1"
                        }
                    ],
                    "walls": [
                        {
                            "assemblyEffectiveRValue": 5,
                            "id": "wall-insulation-1"
                        }
                    ]
                },
                "systems": {
                    "hvac": {
                        "hvacHeatPumps": [
                            {
                                "systemType": "air-to-air",
                                "heatLoadFraction": 1.0,
                                "coolLoadFraction": 1.0,
                                "heatEfficiency": 9.5,
                                "coolEfficiency": 16.0,
                                "backupSystem": {},
                                "id": "heat-pump-systems-1"
                            }
                        ]
                    },
                    "waterHeating": {
                        "waterHeatingSystems": []
                    }
                }
            },
            "globalControls": {
                "debug": true,
                "includeInputs": true,
                "includeBase": true,
                "includeTypical": true,
                "useFixture": false,
                "resolution": [
                    "annual",
                    "monthly"
                ],
                "timelineStartDate": "2024-04-08",
                "timelineDurationInYears": 1,
                "units": {
                    "energyStandard": "MBtu",
                    "coal": "metric ton",
                    "electricity": "MBtu",
                    "fuelOil": "MBtu",
                    "naturalGas": "MBtu",
                    "propane": "MBtu",
                    "woodCord": "MBtu",
                    "woodPellets": "MBtu",
                    "monetary": "USD",
                    "emissions": "CO2"
                }
            }
        }
```

**Response**

```
{
            "globalControls": {
                "timelineStartDate": "2024-04-08",
                "debug": true,
                "includeInputs": true,
                "resolution": [
                    "monthly",
                    "annual"
                ],
                "includeBase": true,
                "includeTypical": true,
                "useFixture": false,
                "timelineDurationInYears": 1,
                "units": {
                    "monetary": "USD",
                    "electricity": "MBtu",
                    "naturalGas": "MBtu",
                    "woodPellets": "MBtu",
                    "propane": "MBtu",
                    "coal": "metric ton",
                    "emissions": "CO2",
                    "woodCord": "MBtu",
                    "energyStandard": "MBtu",
                    "fuelOil": "MBtu"
                }
            },
            "addressFull": "218 S Ohio St, King City, MO 64463-9530",
            "baseBuilding": {
                "systems": {
                    "waterHeating": {
                        "waterHeatingSystems": []
                    },
                    "hvac": {
                        "hvacHeatPumps": [
                            {
                                "heatLoadFraction": 1.0,
                                "heatEfficiency": 9.5,
                                "backupSystem": {},
                                "coolLoadFraction": 1.0,
                                "coolEfficiency": 16.0,
                                "systemType": "air-to-air",
                                "id": "heat-pump-systems-1"
                            }
                        ]
                    }
                },
                "enclosure": {
                    "airInfiltration": {
                        "rate": 2313,
                        "rateUnit": "CFM"
                    },
                    "walls": [
                        {
                            "assemblyEffectiveRValue": 5,
                            "id": "wall-insulation-1"
                        }
                    ],
                    "foundations": [
                        {
                            "type": "crawl unvented",
                            "id": "floor-and-foundation-insulation-1"
                        }
                    ],
                    "attics": [
                        {
                            "floorAssemblyEffectiveRValue": 38,
                            "id": "attic-insulation-1"
                        }
                    ]
                },
                "buildingSummary": {
                    "conditionedFloorArea": 1077,
                    "yearBuilt": 1929,
                    "bedroomsCount": 2,
                    "averageCeilingHeight": 8
                }
            },
            "addressComponents": {
                "zipCode": "64463-9530",
                "addressLine1": "218 S Ohio St",
                "stateAbbreviation": "MO",
                "city": "King City"
            },
            "errors": [
                {
                    "message": "[\"Info: Wrote file: /tmp/modeling-api/openstudio-outputs/BASE/01htzmy7cg2qqvtnqjx7mbt5gp/run/in.xml\" \"Warning: No dishwasher specified, the model will not include dishwasher energy use.\" \"Warning: Assembly R-value (0.7) for 'RimJoistfloor-and-foundation-insulation-1Exposed:0' below minimum expected value. Construction R-value increased to 2.3.\" \"Warning: Assembly R-value (0.7) for 'RimJoistfloor-and-foundation-insulation-1Exposed:90' below minimum expected value. Construction R-value increased to 2.3.\" \"Warning: Assembly R-value (0.7) for 'RimJoistfloor-and-foundation-insulation-1Exposed:180' below minimum expected value. Construction R-value increased to 2.3.\" \"Warning: Assembly R-value (0.7) for 'RimJoistfloor-and-foundation-insulation-1Exposed:270' below minimum expected value. Construction R-value increased to 2.3.\" \"Warning: Assembly R-value (0.7) for 'FoundationWallfloor-and-foundation-insulation-1Exposed' below minimum expected value. Construction R-value increased to 1.32.\" \"Error: Duct location is 'crawlspace - vented' but building does not have this location specified.\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:3585:in `get_space_from_location'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:3561:in `get_space_or_schedule_from_location'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:2437:in `block in create_ducts'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:2432:in `each'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:2432:in `create_ducts'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:2353:in `block in add_airflow'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:2350:in `each'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:2350:in `add_airflow'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:272:in `create'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/measure.rb:118:in `run'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/resources/meta_measure.rb:289:in `run_measure'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/resources/meta_measure.rb:147:in `block (2 levels) in apply_measures'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/resources/meta_measure.rb:139:in `each'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/resources/meta_measure.rb:139:in `block in apply_measures'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/resources/meta_measure.rb:134:in `each'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/resources/meta_measure.rb:134:in `apply_measures'\" \"/var/bin/OpenStudio-HPXML/HPXMLtoOpenStudio/resources/meta_measure.rb:22:in `run_hpxml_workflow'\" \"/var/bin/OpenStudio-HPXML/workflow/run_simulation.rb:40:in `run_workflow'\" \"/var/bin/OpenStudio-HPXML/workflow/run_simulation.rb:157:in `<top (required)>'\" \"eval:121:in `require'\" \"eval:121:in `require'\" \":/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'\" \":/openstudio_cli.rb:1518:in `execute'\" \":/openstudio_cli.rb:761:in `execute'\" \":/openstudio_cli.rb:1755:in `<main>'\" \"eval:175:in `eval'\" \"eval:175:in `require_embedded_absolute'\" \"eval:160:in `block in require_embedded'\" \"eval:154:in `each'\" \"eval:154:in `require_embedded'\" \"eval:113:in `require'\" \"eval:3:in `<main>'\"]",
                    "code": "openstudio-creating-input-unsuccessful"
                }
            ]
        }
```

**Comment by kloconnell** [comment]
@lukeduran Similar to issue #50, this issue has been identified and resolved in the upcoming 2.0.0 release. It is due to the requested `foundation` object not getting applied in all of the appropriate places, causing a mismatch in the duct location logic.

**Comment by kloconnell** [comment]
This request is working in our v2.0.0 development environment.
